### DISCUSSION TITLE
Unable to get inserted HTML through execcommand in editor.getHtml()

### DISCUSSION THREAD

**Discussion by mihir-khandekar** [GrapesJS/grapesjs#3809] [Q&A]
Hello,

I am trying to add an HTML element at the current cursor position, following the instructions mentioned [here](https://github.com/artf/grapesjs/issues/111#issuecomment-310059057) I am able to insert the HTML.

`
editor.Canvas.getBody().ownerDocument.execCommand(
            'insertHTML',
            false,
            `&nbsp;<span class="am-var no-tooltip" data-gjs-editable="false" data-gjs-droppable="false" data-gjs-selectable="false" data-am-var="' +
                key +
                '" style="white-space:nowrap; padding: 1%; font-size: inherit; margin: 1% 2%; cursor: pointer;">' +
                variable +
                '</span>&nbsp;`
        );
`

However when I use the editor API to get the HTML using `editor.getHtml()`, the inserted HTML is not is not part of the returned HTML string.

Please help.

**Comment by artf** [comment]
Try to sync the content of the selected component `editor.getSelected().trigger('sync:content')`

**Reply by mihir-khandekar** [reply]
Tried doing this. isn't working.

**Reply by artf** [reply]
demo?

**Reply by mihir-khandekar** [reply]
I'm using this library as part of the company's product for a new feature. I will need to create a new repo to demo this. Any other way we can go about this? Maybe I can share a larger snippet of code, or Is it possible for us to connect?
